lib/Settings.js

```diff
@@ -42,19 +42,23 @@ Settings.loadRepositoriesSettings = function (configPath) {
     if (repositoryNames.length === 0) {
         return Promise.reject('`repositories` must be non-empty');
     }
-    for (var i = 0; i < repositoryNames.length; i++) {
-        name = repositoryNames[i];
+    return Promise.each(repositoryNames, function (name) {
         if (!/\//.test(name)) {
             return Promise.reject('repository ' + name + ' must be in the form {user}/{repository}');
         }
+
         var repositorySettings = configJson[name];
         if (!defined(repositorySettings.gitHubToken)) {
             return Promise.reject('repository ' + name + ' must have a `gitHubToken`');
         }
 
-        repositories[name] = new RepositorySettings(repositorySettings);
-    }
-    return Promise.resolve();
+        return loadRepoConfig(name, {
```

Comment:
The problem with retrieving everything on startup is that changes made in the repositories will never actually get picked up unless there server is restarted.  I think we need to retrieve the configuration on-demand every time it's needed.  Since this is a very low-traffic service, it shouldn't be that big of a deal and there's no reason to mess with a cache or anything.  We can probably remove most of the nconf usage and refactor the settings to just grab it on demand.